gateway/handler.go

```diff
@@ -640,6 +648,39 @@ func customResponseFormat(r *http.Request) (mediaType string, params map[string]
 	return "", nil, nil
 }
 
+// Add 'Content-Location' headers for non-default response formats. This allows
+// correct caching of such format requests when the format is passed via the
+// Accept header, for example.
+func addContentLocation(r *http.Request, w http.ResponseWriter, rq *requestData) {
+	if rq.responseFormat == "" {
+		return
+	}
+
+	// Response format parameters, such as 'dups' and 'order' for CAR requests
+	// cannot be translated into the URL. Therefore, we cannot add a 'Content-Location'
+	// header.
+	if len(rq.responseParams) != 0 {
+		return
+	}
```

Comment:
:thought_balloon: Hm.. this is tricky.

The reason why we add `Content-Location` is to ensure CAR response triggered via `Accept` header  is not cached under `/ipfs/cid`, as that would be a bug when returned instead of default (deserialized) response type. 

If we skip here, we create a potential DoS hole, which allows clients to maliciously inject CAR with dups as response for non-CAR request.

I see two options here for the value in`Content-Location`:

- (A) update specs and conformance tests and allow for car content type params to be passed in URL query as `format=car&car-version=1&car-dups=n&car-order=dfs`.
- (B) keep this surgical, solve the problem at hand by including `format=car&cache-variant=xxhash(version+dups+order)` if any of car params in `Accept` header was different from the default. This should be enough to nsure response body is not cached under generic `format=car`.

@hacdias maybe do (B) now, and fill issue for (A) in ipfs/specs?

Reply:
@lidel (B) makes me feel uneasy due to the _meaning_ of the `Content-Location` header. If I saw the header response, I would expect that opening such URL would give me the same result as the current request. But that would not be the case here.

(A) likely takes a bit longer, but I think it should be prioritised. I don't think implementing it would take that long, or writing the specs, because:

1. We just ende to add three new query parameters: `car-version`, `car-dups` and `car-order`. Code wise this should be simple. The questions this may rise is: why are the other CAR parameters _not_ part of the `Accept` header (`entity-bytes` and `dag-entity`)?
2. `Accept` header parameter already has priority when both are included, so this is a unnecessary conversation.

Reply:
@lidel see option (A) here: https://github.com/ipfs/boxo/pull/604

Reply:
Thanks! I think this is good enough, we did the due diligence and documented `Content-Location` story in specs, and it is better to have this solution for overwriting different CAR variant in cache, than not. 

To answer the question about `entity-bytes` and `dag-entity` not being parameters of `application/vnd.ipld.car`: these are not CAR specific, nor related to CAR format, but a way to limit the range of DAG being requested. Right now it is only meaningful for CARs, but we could have future format other than CAR that would also use them.